sdks/python/apache_beam/runners/direct/direct_runner.py

```diff
@@ -88,6 +89,11 @@ def run(self, pipeline):
                         evaluation_context)
     # Start the executor. This is a non-blocking call, it will start the
     # execution in background threads and return.
+
+    if pipeline.options:
+      # DirectRunner does not support RuntimeValueProviders.
```

Comment:
Maybe this comment needs to be changed. If options are passed in the command line, they will be of type `StaticValueProvider` and `DirectRunner` can use them. If they were not passed at the command line (or in another way in `PipelineOptions`) prior to calling `run`, there will be no opportunity for the user to tell what the run time values should be. Given that, I do not think `DirectRunner` can support `RuntimeValueProvider`s.